barretenberg/cpp/src/barretenberg/honk/proof_system/ultra_prover.cpp

```diff
@@ -156,16 +156,118 @@ template <UltraFlavor Flavor> void UltraProver_<Flavor>::execute_univariatizatio
 template <UltraFlavor Flavor> void UltraProver_<Flavor>::execute_pcs_evaluation_round()
 {
     const FF r_challenge = transcript.get_challenge("Gemini:r");
-    gemini_output = Gemini::compute_fold_polynomial_evaluations(
+    univariate_openings = Gemini::compute_fold_polynomial_evaluations(
         sumcheck_output.challenge_point, std::move(gemini_polynomials), r_challenge);
 
     for (size_t l = 0; l < instance->proving_key->log_circuit_size; ++l) {
         std::string label = "Gemini:a_" + std::to_string(l);
-        const auto& evaluation = gemini_output.opening_pairs[l + 1].evaluation;
+        const auto& evaluation = univariate_openings.opening_pairs[l + 1].evaluation;
         transcript.send_to_verifier(label, evaluation);
     }
 }
 
+/**
+ * @brief Prove proper construction of the aggregate Goblin ECC op queue polynomials T_i^(j), j = 1,2,3,4.
+ * @details Let T_i^(j) be the jth column of the aggregate op queue after incorporating the contribution from the
+ * present circuit. T_{i-1}^(j) corresponds to the aggregate op queue at the previous stage and $t_i^(j)$ represents
+ * the contribution from the present circuit only. For each j, we have the relationship T_i = T_{i-1} + right_shift(t_i,
+ * M_{i-1}), where the shift magnitude M_{i-1} is the length of T_{i-1}. This stage of the protocol demonstrates that
+ * the aggregate op queue has been constructed correctly.
+ *
+ */
+template <UltraFlavor Flavor> void UltraProver_<Flavor>::execute_op_queue_transcript_aggregation_round()
+{
+    if constexpr (IsGoblinFlavor<Flavor>) {
+        // Extract size M_{i-1} of T_{i-1} from op_queue
+        size_t prev_op_queue_size = instance->proving_key->op_queue->get_previous_size(); // M_{i-1}
+        // TODO(#723): Cannot currently support an empty T_{i-1}. Need to be able to properly handle zero commitment.
+        ASSERT(prev_op_queue_size > 0);
+
+        auto circuit_size = instance->proving_key->circuit_size;
+
+        // TODO(#723): The below assert ensures that M_{i-1} + m_i < n, i.e. the right shifted result can be expressed
+        // as a size n polynomial. If this is not the case then we should still be able to proceed without increasing
+        // the circuit size but need to handle with care.
+        ASSERT(prev_op_queue_size + instance->proving_key->num_ecc_op_gates < circuit_size); // M_{i-1} + m_i < n
```

Comment:
This should not be an issue. You should only care that the current ecc op gates fit in the circuit size, but the previous ones should have no influence on the circuit size. Otherwise we'd have to make all the circuits the same size, since you never know how much of the queue you have to fit.

Reply:
Right I agree. I made it an issue to deal with later because I think some care needs to be taken since we have not historically allowed polynomials with size > `circuit_size` to be floating around. Just want to make sure there are no footguns